include/dxc/Support/dxcapi.extval.h

```diff
@@ -0,0 +1,56 @@
+#include "dxc/Support/dxcapi.use.h"
+#include <cassert>
+#include <string>
+
+namespace dxc {
+class DxcDllExtValidationLoader : public DllLoader {
+  // DxCompilerSupport manages the
+  // lifetime of dxcompiler.dll, while DxilExtValSupport
+  // manages the lifetime of dxil.dll
+  dxc::SpecificDllLoader DxCompilerSupport;
+  dxc::SpecificDllLoader DxilExtValSupport;
+
+  std::string DxilDllPath;
+  HRESULT InitializeInternal(LPCSTR fnName);
+
+public:
+  std::string GetDxilDllPath() { return DxilDllPath; }
+  bool DxilDllFailedToLoad() {
+    return !DxilDllPath.empty() && !DxilExtValSupport.IsEnabled();
+  }
+
+  HRESULT CreateInstanceImpl(REFCLSID clsid, REFIID riid,
+                             IUnknown **pResult) override;
+  HRESULT CreateInstance2Impl(IMalloc *pMalloc, REFCLSID clsid, REFIID riid,
+                              IUnknown **pResult) override;
+
+  HRESULT Initialize() { return InitializeInternal("DxcCreateInstance"); }
+
+  /* Note, OverrideDll takes this dll argument and ignores it
+  to satisfy the IDllLoader interface. The parameter is ignored
+  because the relevant dlls are specific and known: dxcompiler.dll
+  and dxil.dll. This class is not designed to handle any other dlls */
+  HRESULT OverrideDll(LPCSTR dll, LPCSTR entryPoint) override {
+    return InitializeInternal(entryPoint);
+  }
```

Comment:
I'm not convinced this has to be a part of the `DllLoader` interface either, it'd be nice to remove it because it clearly doesn't really fit in neatly.

I've been doing some searching and every place I've found that calls `OverrideDll` can be traced back to a concrete `SpecificDllLoader`.